File: modified_scripts/c++/binary_search/qwen3_1.7b.cpp
#include <algorithm>

template <typename Iter>
void binary_search(Iter first, Iter last, typename std::iterator_traits<Iter>::value_type value) {
    while (first < last) {
        Iter mid = first + (last - first) / 2;
        if (*mid < value) {
            first = mid + 1;
        } else {
            last = mid;
        }
    }
    if (*first < value) {
        first = last;
    }
}

File: modified_scripts/julia/binary_search/qwen3_1.7b.jl
function binary_search(arr, x)
    left = 1
    right = length(arr)
    while left <= right
        mid = (left + right) ÷ 2
        if arr[mid] == x
            return mid
        elseif arr[mid] < x
            left = mid + 1
        else
            right = mid - 1
        end
    end
    return -1
end